File: valid-anagram/valid-anagram.py
class Solution:
    def isAnagram(self, s: str, t: str) -> bool:
        dic = {}
        for char in s:
            if not char in dic:
                dic[char] = 0
            dic[char] += 1
        
        for char in t:
            if not char in dic:
                return False
            dic[char] -= 1
        
        for value in dic.values():
            if value != 0:
                return False
        
        return True

File: valid-anagram/valid-anagram.cs
public class Solution {
    public bool IsAnagram(string s, string t) {
        if (s.Length != t.Length) return false;
        var dic = new Dictionary<char, int>();
        
        foreach (char c in s)
        {
            if (!dic.ContainsKey(c)) dic.Add(c, 0);
            dic[c] += 1;
        }
        
        foreach (char c in t)
        {
            if (!dic.ContainsKey(c)) return false;
            dic[c] -= 1;
        }
        
        return dic.Values.All(x => x == 0);
    }
}